src/contexts/signup-context.js

```diff
@@ -0,0 +1,71 @@
+import axios from "axios";
+import { createContext, useContext, useReducer, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useLogin } from "./login-context";
+
+const SignupContext = createContext("");
+
+const defaultSignUpdetailsTemplate = {
+    email: "",
+    firstName: "",
+    lastName: "",
+    password: "",
+    contact: "",
+}
+
+const signupReducer = (signupDetails, action) => {
+    switch(action.type) {
+        case "EMAIL" :
+            return {...signupDetails, email : action.payload}
+        case "PASSWORD" : 
+            return {...signupDetails, password : action.payload}
+        case "FIRST_NAME" : 
+            return {...signupDetails, firstName : action.payload}
+        case "LAST_NAME" : 
+            return {...signupDetails, lastName : action.payload}
+        case "PHONE_NO" : 
+            return {...signupDetails, contact : action.payload}
+        default: return ({...defaultSignUpdetailsTemplate})
+    }
+}
```

Comment:
Do shift it to another location to make it easy to use later on!!
Meanwhile in the reducer you can destructure it on the spot itself:
const signupReducer=(signupDetails,{type,payload})=>{

}
If it doesn't works then you can continue to use this approach